rtmp-server/src/handshake.rs

```diff
@@ -0,0 +1,57 @@
+use std::{
+    io::{Read, Write},
+    time::Instant,
+};
+
+use crate::error::RtmpError;
+use rand::RngCore;
+
+const RTMP_VERSION: u8 = 3;
+const HANDSHAKE_SIZE: usize = 1536;
+
+pub struct Handshake;
+
+impl Handshake {
+    pub fn perform<S>(stream: &mut S) -> Result<(), RtmpError>
+    where
+        S: Read + Write,
+    {
+        // C0 version
+        let mut c0 = [0u8; 1];
+        stream.read_exact(&mut c0)?;
+        let c0_read_time = Instant::now();
+
+        // S0 version
+        stream.write_all(&[RTMP_VERSION])?;
+
+        // S1 timestamp(4 bytes), zero(4 bytes), random(1528 bytes)
+        let mut s1 = [0u8; HANDSHAKE_SIZE];
+        let timestamp: u32 = 0;
+        s1[0..4].copy_from_slice(&timestamp.to_be_bytes());
+        s1[4..8].copy_from_slice(&[0u8; 4]); // zeros
```

Comment:
```suggestion
        s1[4..8].copy_from_slice(&[0; 4]);
```

I am not 100% sure, but it should work without explicitly typing the literal here